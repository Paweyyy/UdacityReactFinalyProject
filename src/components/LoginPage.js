import * as React from 'react';
import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import { setAuthedUser } from '../actions/authedUser';

const LoginPage = ({ users, dispatch }) => {
  const navigate = useNavigate();

  const handleSelect = e => {
    if(e.target.value !== ""){
      dispatch(setAuthedUser(e.target.value));
      navigate("/")
    }
  }

  return (
    <div className='pageContainer loginContainer'>
      <h1 className=''>Employee Polls</h1>
      <img alt="Login Img" className='loginImg' src='employee.jpeg'/>
      <p>Please select your username:</p>
      <select onChange={handleSelect} name="pets" id="pet-select">
          <option value="">--Please choose your username--</option>
          {users.map(user => {
            return <option value={user}>{user}</option>
          })
          }
      </select>


    </div>
  );
}

const mapStateToProps = ({ users, dispatch }) => {
  return {users: Object.keys(users), dispatch}
}

export default connect(mapStateToProps)(LoginPage);