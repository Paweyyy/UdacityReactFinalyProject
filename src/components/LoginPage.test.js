import { render, fireEvent,find } from "@testing-library/react";
import * as React from 'react';
import LoginPage from "./LoginPage";
import { createStore } from "redux";
import { Provider } from "react-redux";
import reducer from "../reducers";
import { BrowserRouter as Router } from "react-router-dom";
import '@testing-library/jest-dom/extend-expect';
import thunk from "redux-thunk";
import { applyMiddleware } from "redux";

describe("LoginPage", () => {
    it("will match LoginSnapshot", () => {
        const middleware = applyMiddleware(thunk)

        const store = createStore(reducer, middleware);
    
        var component = render(<Provider store={store}><Router><LoginPage /></Router></Provider>);
    
        expect(component).toMatchSnapshot();
    })
})