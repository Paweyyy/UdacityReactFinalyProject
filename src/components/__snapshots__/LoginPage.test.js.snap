// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage will match LoginSnapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pageContainer loginContainer"
      >
        <h1
          class=""
        >
          Employee Polls
        </h1>
        <img
          alt="Login Img"
          class="loginImg"
          src="employee.jpeg"
        />
        <p>
          Please select your username:
        </p>
        <select
          data-testid="username-selection"
          id="pet-select"
          name="pets"
        >
          <option
            value=""
          >
            --Please choose your username--
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pageContainer loginContainer"
    >
      <h1
        class=""
      >
        Employee Polls
      </h1>
      <img
        alt="Login Img"
        class="loginImg"
        src="employee.jpeg"
      />
      <p>
        Please select your username:
      </p>
      <select
        data-testid="username-selection"
        id="pet-select"
        name="pets"
      >
        <option
          value=""
        >
          --Please choose your username--
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
